### PR TITLE
Enable Transparent Hugepage on xLinux by default

### PR DESCRIPTION
**Pull request by fengxue-IS**
Relates: #6156 

Signed-off-by: Jack Lu <Jack.S.Lu@ibm.com>

### CODE REVIEW

**Comment by fengxue-IS** [comment]
@DanHeidinga can I get a review please

@vijaysun-omr could the perf team please look into this change and verify the performance impact please

**Comment by DanHeidinga** [comment]
Given the results listed in #6156, I'm OK with proceeding with this though I'd appreciate any additional footprint results on X that can be shared.

File: runtime/vm/jvminit.c [hunk 1]
```diff
@@ -1821,7 +1821,12 @@ IDATA VMInitStages(J9JavaVM *vm, IDATA stage, void* reserved) {
 				if (argIndex2 > argIndex) {
 					j9port_control(J9PORT_CTLDATA_VMEM_ADVISE_HUGEPAGE, 1);
 				} else {
+#if defined(LINUX) && !defined(LINUXPPC) && !defined(J9ZTPF)
```
Comment (DanHeidinga) [inline runtime/vm/jvminit.c]:
```suggestion
#if defined(LINUX) && defined(J9VM_ARCH_X86)
```

File: runtime/vm/jvminit.c [hunk 2]
```diff
@@ -1821,7 +1821,12 @@ IDATA VMInitStages(J9JavaVM *vm, IDATA stage, void* reserved) {
...
+					/* Enable THP on xLinux by default */
+					j9port_control(J9PORT_CTLDATA_VMEM_ADVISE_HUGEPAGE, 1);
+#else
 					j9port_control(J9PORT_CTLDATA_VMEM_ADVISE_HUGEPAGE, 0);
+#endif /* defined(LINUX) && !defined(LINUXPPC) && !defined(J9ZTPF) */
```
Comment (DanHeidinga) [inline runtime/vm/jvminit.c]:
```suggestion
#endif /* defined(LINUX) && defined(J9VM_ARCH_X86) */
```

**Comment by piyush286** [comment]
We are going to launch some runs to verify the performance impact. Will post the results once we have them.

**Comment by piyush286** [comment]
@fengxue-IS Could you please provide us with the build? 

@DanHeidinga Just to confirm, we want to compare the perf between THP (i.e. no option/default or XX:+TransparentHugePage) and no THP (i.e. XX:-TransparentHugePage)?

FYI @dhong44

**Comment by dhong44** [comment]
I think you may have tagged the wrong person here @piyush286

**Comment by fengxue-IS** [comment]
@piyush286 I'll send you the build link shortly, We are looking to compare  THP (i.e. no option/default or XX:+TransparentHugePage) and no THP (i.e. XX:-TransparentHugePage) on xLinux.

@DanHeidinga I've noticed a issue with my change, it will override the user option for `-XX:-TransparentHugePage` on x, added a new commit to fix this, can you please do a re-review. Thanks

**Comment by DanHeidinga** [comment]
Jenkins test sanity plinux,xlinux jdk8

**Comment by DanHeidinga** [comment]
@fengxue-IS Any update on the performance testing?

**Comment by dhlee49** [comment]
Benchmark result shows that enabling THP does not introduce any regression

**Liberty Test result**

LibertyDT3 | With THP | With No THP | THP / no THP
-- | -- | -- | --
Throughput | 10221.55 | 10181.44 | 100.39%
CI | 0.53% | 0.68% |  
Footprints | 1215.35742 | 1214.46953 | 99.93%
CI | 0.14% | 0.19% |  
LibertyDT3Warm |   |   |  
Throughput | 10212.3667 | 10192.75 | 100.19%
CI | 0.63% | 0.87% |  
Footprints | 1200.48112 | 1200.48535 | 100.00%
CI | 0.16% | 0.11% |  
LibertyDT7 |   |   |  
Throughput | 5102.26667 | 5070.21667 | 100.63%
CI | 1.17% | 0.63% |  
Footprints | 834.259766 | 831.402995 | 99.66%
CI | 0.52% | 0.33% |  
LibertyDT7 |   |   |  
Throughput | 5063.88333 | 5099.68333 | 99.30%
CI | 1.48% | 1.05% |  
Footprints | 814.098307 | 813.400391 | 99.91%
CI | 0.70% | 0.86% |  



**SPECjbb test result**

  | Without THP | With THP |  
-- | -- | -- | --
max_jOPS | 39270 | 39093 | 99.55%
CI | 5.73% | 0.00% |  
critical_jOPS | 13712 | 13481.5 | 98.32%
CI | 35.77% | 12.49% |

**Comment by DanHeidinga** [comment]
FYI @SueChaplain Might be good to call this out in the release notes for `0.16.0`

**Comment by SueChaplain** [comment]
@DanHeidinga - we should have an issue for this at openj9-docs as the main user doc set will need updating too: https://www.eclipse.org/openj9/docs/xxtransparenthugepage/

Currently shows as disabled by default on x86, POWER AND Z Linux. 

Are the other platforms expecting to enable by default too? 
@pmhayward FYI

**Comment by DanHeidinga** [comment]
Opened https://github.com/eclipse/openj9-docs/issues/340 for the docs

**Pull request merged**